case_studies/sdss_galaxies/plots.py

```diff
@@ -142,38 +143,34 @@ def fignames(self):
             "classification": "sdss-classification-acc.png",
         }
 
-    def compute_data(self, scene, coadd_cat, sleep_net, binary_encoder, galaxy_encoder):
+    def compute_data(self, scene, coadd_cat, encoder, decoder):
         assert isinstance(scene, (torch.Tensor, np.ndarray))
-        assert sleep_net.device == binary_encoder.device == galaxy_encoder.device
-        device = sleep_net.device
+        assert encoder.device == decoder.device
+        device = encoder.device
 
         bp = 24
-        clen = 300
+        slen = 300  # chunk side-length
         h, w = scene.shape[-2], scene.shape[-1]
 
+        # FIXME: For now reconstruct_scene_at_coordinates can only handle square images.
+        # so we use min(h,w), remove once we can deal with rectangular images.
+        scene_length = min(h, w) + 2 * bp
+
         # load coadd catalog
-        coadd_params = reporting.get_params_from_coadd(coadd_cat, h, w, bp)
+        coadd_params = reporting.get_params_from_coadd(
+            coadd_cat, xlim=(bp, scene_length - bp), ylim=(bp, scene_length - bp)
+        )
 
         # misclassified galaxies in PHOTO as galaxies (obtaind by eye)
         ids = [8647475119820964111, 8647475119820964100, 8647475119820964192]
         for my_id in ids:
             idx = np.where(coadd_params["objid"] == my_id)[0].item()
             coadd_params["galaxy_bools"][idx] = 0
 
-        # load specific models that are needed.
-        image_encoder = sleep_net.image_encoder.to(device).eval()
-        galaxy_decoder = sleep_net.image_decoder.galaxy_tile_decoder.galaxy_decoder.eval()
-
         # predict using models on scene.
         scene_torch = torch.from_numpy(scene).reshape(1, 1, h, w)
-        _, est_params = predict_on_scene(
-            clen,
-            scene_torch,
-            image_encoder,
-            binary_encoder,
-            galaxy_encoder,
-            galaxy_decoder,
-            device,
+        _, est_params = reconstruct_scene_at_coordinates(
+            encoder, decoder, scene_torch, bp, bp, scene_length, slen, bp, device=device
```

Comment:
I think this will work, but the intention of `reconstruct_scene_at_coordinates` is so you don't have to worry about border padding - you give it the image coordinates (h, w, scene_length) and it will give you back the reconstructed image and parameters.